src/package/midlrt/Microsoft.Windows.MidlRT.targets

```diff
@@ -0,0 +1,339 @@
+            MidlRTCopyWinMDToOutputDirectory;
+        </AfterMidlTargets>
+        <ResolveAssemblyReferencesDependsOn>
+            $(ResolveAssemblyReferencesDependsOn);GetMidlRTProjectWinMDReferences;MidlRTMarkStaticLibrariesPrivate;
+        </ResolveAssemblyReferencesDependsOn>
+        <CleanDependsOn>
+            $(CleanDependsOn);MidlRTClean
+        </CleanDependsOn>
+
+    </PropertyGroup>
+
+    <!-- For a static library we don't want the winmd/lib/pdb to be packaged -->
+    <PropertyGroup Condition="'$(ConfigurationType)' == 'StaticLibrary'">
+        <IncludeCopyWinMDArtifactsOutputGroup>false</IncludeCopyWinMDArtifactsOutputGroup>
+    </PropertyGroup>
+
+    <Target Name="MidlRTClean">
+        <ItemGroup>
+            <_FilesToDelete Remove="@(_FilesToDelete)"/>
+            <_FilesToDelete Include="$(OutDir)*.winmd"/>
+            <_FilesToDelete Include="$(IntDir)*.winmd"/>
+            <_FilesToDelete Include="$(IntDir)*.rsp"/>
+            <_FilesToDelete Include="$(MidlRTMergedDir)**"/>
+            <_FilesToDelete Include="$(MidlRTUnmergedDir)**"/>
+        </ItemGroup>
+        <Delete Files="@(_FilesToDelete)"/>
+    </Target>
+
+    <!-- Target used only to evaluate MidlRTGenerateWindowsMetadata if it doesn't already have a value -->
+    <Target Name="MidlRTComputeGenerateWindowsMetadata"
+            DependsOnTargets="GetMidlRTProjectWinMDReferences;$(MidlRTComputeGenerateWindowsMetadataDependsOn)">
+
+        <PropertyGroup>
+            <!-- For static libraries, only idl causes a winmd to be generated. 
+                 For exe/dll, static libraries that produce a WinMD will be merged, 
+                 so they also cause a WinMD to be generated-->
+            <MidlRTGenerateWindowsMetadata Condition="'$(ConfigurationType)' != 'StaticLibrary' AND '@(MidlRTStaticProjectWinMDReferences)@(Midl)'!= ''">true</MidlRTGenerateWindowsMetadata>
+            <MidlRTGenerateWindowsMetadata Condition="'$(ConfigurationType)' == 'StaticLibrary' AND '@(Midl)'!= ''">true</MidlRTGenerateWindowsMetadata>
+
+            <!-- At this point we checked all cases where we do generate a WinMD.
+                 The remaining option is no WinMD. -->
+            <MidlRTGenerateWindowsMetadata Condition="'$(MidlRTGenerateWindowsMetadata)'== ''">false</MidlRTGenerateWindowsMetadata>
+        </PropertyGroup>
+
+    </Target>
+
+    <Target Name="ComputeGetResolvedWinMD"
+            Condition="'$(MidlRTGenerateWindowsMetadata)' == ''">
+        <!-- If MidlRTGenerateWindowsMetadata is not defined, compute it.-->
+        <!-- We use Calltarget, so we don't run anything including DependsOnTargets
+             targets if $(MidlRTGenerateWindowsMetadata) already has a value.-->
+        <CallTarget Targets="MidlRTComputeGenerateWindowsMetadata" />
+    </Target>
+
+    <!-- This target overrides the GetResolvedWinMD target used to resolve the WinMD for native projects
+         so it is aware of the C++/WinRT generated WinMD.
```

Comment:
any comments mentioning "C++/WinRT" should be reviewed.

Reply:
good catch - we don't need `MidlRTGenerateWindowsMetadata` because that's the only thing this package is for :)